### DISCUSSION TITLE
Welcome to BCC Discussions!

### DISCUSSION THREAD

**Discussion by chenhengqi** [iovisor/bcc#4330] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by tahirali-csc** [comment]
In `tcplife.bpf.c`, when reading ip4/pi6 source/dest addresses  https://github.com/iovisor/bcc/blob/8422cd449ad2e60414a4508aa4a150a9db39c4a3/libbpf-tools/tcplife.bpf.c#L138, we use `bpf_probe_read_kernel` instead of `bpf_core_read`. The whole purpose of  `bpf_core_read` is that handles fields offset changes transparently across different kernel version. Any thoughts?

**Reply by chenhengqi** [reply]
Please open a new discussion.
